diff --git a/termite.cc b/termite.cc
index bdb01e3..e089958 100644
--- a/termite.cc
+++ b/termite.cc
@@ -847,6 +847,22 @@ gboolean key_press_cb(VteTerminal *vte, GdkEventKey *event, keybind_info *info)
             }
             return TRUE;
         }
+        if (modifiers == GDK_MOD1_MASK) {
+            switch (gdk_keyval_to_lower(event->keyval)) {
+                case GDK_KEY_k:
+                    move(vte, &info->select, 0, -(vte_terminal_get_row_count(vte) / 2));
+                    break;
+                case GDK_KEY_j:
+                    move(vte, &info->select, 0, vte_terminal_get_row_count(vte) / 2);
+                    break;
+                case GDK_KEY_h:
+                    move_backward_word(vte, &info->select);
+                    break;
+                case GDK_KEY_l:
+                    move_forward_word(vte, &info->select);
+                    break;
+            }
+        }
         if (modifiers == GDK_SHIFT_MASK) {
             switch (event->keyval) {
                 case GDK_KEY_Left:
@@ -858,8 +874,11 @@ gboolean key_press_cb(VteTerminal *vte, GdkEventKey *event, keybind_info *info)
             }
         }
         switch (event->keyval) {
+            case GDK_KEY_i:
+            case GDK_KEY_a:
             case GDK_KEY_Escape:
             case GDK_KEY_q:
+                move_to_row_start(vte, &info->select, last_row(vte));
                 exit_command_mode(vte, &info->select);
                 gtk_widget_hide(info->panel.da);
                 gtk_widget_hide(info->panel.entry);
@@ -938,6 +957,11 @@ gboolean key_press_cb(VteTerminal *vte, GdkEventKey *event, keybind_info *info)
 #else
                 vte_terminal_copy_clipboard(vte);
 #endif
+                move_to_row_start(vte, &info->select, last_row(vte));
+                exit_command_mode(vte, &info->select);
+                gtk_widget_hide(info->panel.da);
+                gtk_widget_hide(info->panel.entry);
+                info->panel.url_list.clear();
                 break;
             case GDK_KEY_slash:
                 overlay_show(&info->panel, overlay_mode::search, vte);
@@ -976,6 +1000,23 @@ gboolean key_press_cb(VteTerminal *vte, GdkEventKey *event, keybind_info *info)
         }
         return TRUE;
     }
+    if (modifiers == GDK_MOD1_MASK) {
+        switch (event->keyval) {
+            case GDK_KEY_f:
+                enter_command_mode(vte, &info->select);
+                find_urls(vte, &info->panel);
+                gtk_widget_show(info->panel.da);
+                overlay_show(&info->panel, overlay_mode::urlselect, nullptr);
+                exit_command_mode(vte, &info->select);
+                return TRUE;
+            case GDK_KEY_p:
+                vte_terminal_paste_clipboard(vte);
+                return TRUE;
+            case GDK_KEY_y:
+                enter_command_mode(vte, &info->select);
+                return TRUE;
+        }
+    }
     if (modifiers == (GDK_CONTROL_MASK|GDK_SHIFT_MASK)) {
         switch (gdk_keyval_to_lower(event->keyval)) {
             case GDK_KEY_plus:
